text/XXXX-config_revision.md

```diff
@@ -0,0 +1,268 @@
+# Summary
+[summary]: #summary
+
+Define a hierarchy for configuration overrides, revise configuration fields with clear descriptions.
+
+# Motivation
+[motivation]: #motivation
+
+Our configuration story needs work. _Many_ fields are not in use, and there's ambiguity about how 
+
+This RFC proposes a few concrete changes:
+1. Overhaul configuration data structures, removing unnecessary fields & adding new ones
+1. Define how environment variables and configuration interact.
+1. Formalize the hierarchy of different configuration sources
+1. Formalize the list of enviornment variables Qri is sensitive to.
+1. Switch to multiaddresses for all network configuration.
+1. Move ipfs repo location into configuration, deprecate support for `IPFS_PATH`.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+
+### Environment Variables don't configure, they point to configuration files
+
+This is the total list of environemnt variables qri is sensitive to:
+
+| variable name            | description                           |
+| ------------------------ | ------------------------------------- |
+| `PAGER`                  | what pager to use for output          |
+| `QRI_PATH`               | path to qri configuraiton data        |
+| `QRI_SETUP_CONFIG_DATA`  | JSON data to use when creating a repo |
+| `IPFS_SETUP_CONFIG_DATA` | JSON data when initializing IPFS      |
+| `QRI_PRIVATE_KEY`        | defined in the private key RFC        |
+| `QRI_BACKTRACE`          | show stack trace output on crash      |
+
+
+_note: I (b5) haven't had a chance to vet our dependencies for env vars they are sensitive to, so this list is only complete for the purpose of this RFC._
+
+This RFC proposes environment variables are _not_ a direct source of configuration. Instead environment variables either point to a path to load configuration from, or contain a payload to create an initial configuration file.
+
+`QRI_SETUP_CONFIG_DATA` as the only supported is only honored by `qri setup`. We need `SETUP` env variables for operating within containerized contexts, but that's about it.
+
+Directing as much configuration as possible to a configuration file cuts down on potential sources of confusion. As an example, `git` will listen to `$PAGER`, `$GIT_PAGER`, and `git config core.pager`, in that order, and I (b5) had to look up what the order is.
+
+Instead of supporting a potential `QRI_PAGER` environment variable, we should _only_ have `qri.cli.pager` to override the default value.
+
+All that said, using env var configuration should be _optional_, meaning a user employing qri as a library can provide an option to the `lib.NewInstance` constructor function that ignores all environment variables.
+
+### Config Hierarchy
+
+Configuration details should be unified into a single data structure loaded in a predefined hierarchy. From least that hierarchy should be:
+
+* default configuration
+* qri repo configuration file
+* global command-line flags
+
```

Comment:
I'd like to add that ENV vars should still remain in this hierarchy and that it's ok for an env var to specify a config and not just point to some. However it's context is and should be limited to the session itself and as such is pretty high up in the hierarchy as it's typically treates as a user supplied default without neccesarily materializing this change into the config.

As such I'd squeze it in between repo config and cmd flags.